### ISSUE TITLE
[a11y]: UI Shell right panel does not close on loss of focus

### ISSUE THREAD

**Issue by annawen1** [carbon-design-system/carbon#19289]
### Package

@carbon/react, @carbon/web-components

### Package version

latest

### Automated testing tool and ruleset

n/a

### Description

The right panel of the UI Shell does not seem to close when the associated icon loses focus, which ends up covering up other content.

https://medium.com/design-ibm/assessing-components-and-patterns-for-wcag-2-2-47ee3f0f468e
https://www.w3.org/TR/WCAG22/#focus-not-obscured-minimum

### WCAG 2.1 Violation

2.4.11 Focus Not Obscured (Minimum)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-actions-and-right-panel

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-actions-and-right-panel
2. Notice when focusing on elements on the page, the right panel does not go away and ends up covering the content on the page

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
